app/src/main/java/org/dslul/openboard/inputmethod/latin/spellcheck/AndroidWordLevelSpellCheckerSession.java

```diff
@@ -110,12 +112,47 @@ public void onChange(boolean self) {
         cres.registerContentObserver(Words.CONTENT_URI, true, mObserver);
     }
 
+    private void updateLocale() {
+        final String localeString = getLocale();
+
+        if (mLocale == null || !mLocale.toString().equals(localeString)) {
+            final String oldLocal = mLocale == null ? "null" : mLocale.toString();
+            Log.d(TAG, "Updating locale from " + oldLocal + " to " + localeString);
+
+            mLocale = (null == localeString) ? null
+                    : LocaleUtils.constructLocaleFromString(localeString);
+            mScript = ScriptUtils.getScriptFromSpellCheckerLocale(mLocale);
+        }
+    }
+
     @Override
     public void onCreate() {
-        final String localeString = getLocale();
-        mLocale = (null == localeString) ? null
-                : LocaleUtils.constructLocaleFromString(localeString);
-        mScript = ScriptUtils.getScriptFromSpellCheckerLocale(mLocale);
+        updateLocale();
+    }
+
+    @Override
+    public String getLocale() {
+        // This function was taken from https://github.com/LineageOS/android_frameworks_base/blob/1235c24a0f092d0e41fd8e86f332f8dc03896a7b/services/core/java/com/android/server/TextServicesManagerService.java#L544 and slightly adopted.
+
+        final InputMethodManager imm;
```

Comment:
@MajeurAndroid I guess one could move the initialization of `imm` into onCreate because that probably won't change. But that shouldn't be a huge performance improvement.